src/Common/src/TypeSystem/IL/Stubs/AddrOfIntrinsic.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Internal.TypeSystem;
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.IL.Stubs
+{
+    /// <summary>
+    /// Provides method bodies for AddrOf intrinsics. The intrinsics work around the inability to express
+    /// ldftn instruction in C#. The intrinsic method returns the target address of delegate pointing to 
+    /// static method. The assumption is that the delegate was just initialized and thus the optimizing
+    /// code generator will be able to eliminate the unnecessary delegate allocation.
+    /// </summary>
+    public static class AddrOfIntrinsic
+    {
+        public static MethodIL EmitIL(MethodDesc target)
+        {
+            Debug.Assert(target.Name == "AddrOf");
+            Debug.Assert(target.Signature.Length == 1
+                && target.Signature.ReturnType == target.Context.GetWellKnownType(WellKnownType.IntPtr));
+
+            ILEmitter emitter = new ILEmitter();
+            var codeStream = emitter.NewCodeStream();
+
+            codeStream.EmitLdArg(0);
+
+            var fptrField = target.Context.GetWellKnownType(WellKnownType.MulticastDelegate).BaseType.GetField("m_extraFunctionPointerOrData");
```

Comment:
Maybe. This is a second place with this pattern - I am not sure how representative these two cases are. I look at these as hacks to make to make progress - that would better done differently. E.g. With the current state of affairs, AddrOf does not actually need to be an intrinsic with hand-emitted IL at all. It can be a regular method somewhere, that just recognized as intrinsic by optimizations.